### PR TITLE
Source twilio: fix generating date ranges

### PR DESCRIPTION
**Pull request by davydov-d**
## What
https://github.com/airbytehq/oncall/issues/1046

## How
When generating datetime ranges we should take into account the cursor field type - in case it is of `date` type, the stream used to yield extra slice where start date was greater than end date. Now this is fixed

### CODE REVIEW

**Review by girarda — APPROVED** [review]

File: airbyte-integrations/connectors/source-twilio/acceptance-test-config.yml [hunk 1]
```diff
@@ -1,35 +1,53 @@
...
+          - name: "conference_participants"
+            bypass_reason: "stream not filled yet"
+          - name: "available_phone_numbers_local"
+            bypass_reason: "very volatile data"
+          - name: "available_phone_numbers_mobile"
+            bypass_reason: "very volatile data"
+          - name: "available_phone_numbers_toll_free"
+            bypass_reason: "very volatile data"
+          - name: "usage_records"
+            bypass_reason: "very volatile data"
```
Comment (alafanechere) [inline airbyte-integrations/connectors/source-twilio/acceptance-test-config.yml:35]: I logged an issue to get better seeding on this account: https://github.com/airbytehq/airbyte/issues/19876

**Review by alafanechere — APPROVED** [review]
Approving changes made to `airbyte-integrations/connectors/source-twilio/acceptance-test-config.yml` on behalf of the connector operations team.

**Pull request merged**